File: 20220915/모범답안/20220915_09.js/follower.py
class FriendShip():
    def __init__(self):
        self.friendMap = dict()
        self.fff = 0

    def setFriendShip(self, friend):
        try:
            self.friendMap[friend[0]].add(friend[1])
        except KeyError:
            self.friendMap[friend[0]] = set([friend[1]])

        if friend[1] not in self.friendMap or friend[0] not in self.friendMap[friend[1]]:
            return
        else:
            self.fff += 1
        return


def solution(friends):
    fs = FriendShip()

    for friend in friends:
        fs.setFriendShip(friend)

    return fs.fff

File: 20220915/모범답안/20220915_09.js/follower.js
function solution(friends)
{
  var map = new Map();
  var result = 0;
  var target = [];

  for (var i = 0; i < friends.length; i++)
  {
    target = map.get(friends[i][1]);
    if (target == undefined || target.indexOf(friends[i][0]) == -1)
    {
      target = target == undefined ? [friends[i][1]] : target.push(friends[i][1]);
      map.set(friends[i][0], target);
    }
    else
    {
      result ++;
    }
  }

  return result;
}